code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -105,6 +105,9 @@
 		if(!istype(I, /obj/item/clothing))
 			var/final_block_chance = I.block_chance - (CLAMP((armour_penetration-I.armour_penetration)/2,0,100)) + block_chance_modifier //So armour piercing blades can still be parried by other blades, for example
 			if(I.hit_reaction(src, AM, attack_text, final_block_chance, damage, attack_type))
+				var/obj/item/shield/S = I
+				if (istype(S))

```

Comment:
shouldn't this be flipped?

You should check the type to ensure it's a shield BEFORE typecasting it as such.